web-components/src/components/table-advanced/TableAdvanced.ts

```diff
@@ -0,0 +1,994 @@
+            type: "select",
+            index: i
+          });
+        }
+      }
+
+      this.requestUpdate("selected");
+    }
+
+    private clearSelection() {
+      this.selected = {};
+      this.requestUpdate("selected");
+    }
+
+    private collapseToggle(e: Event, row: number) {
+      e.stopPropagation();
+      if (this.expandedRowIdx[row]) {
+        delete this.expandedRowIdx[row];
+        this["md-table-advanced-change"].emit({
+          type: "collapse",
+          row
+        });
+      } else {
+        this.expandedRowIdx[row] = true;
+        this["md-table-advanced-change"].emit({
+          type: "expand",
+          row
+        });
+      }
+      this.clearSelection();
+      this.requestUpdate();
+    }
+
+    private eX = 0; // https://stackoverflow.com/a/33672982
+
+    // RESIZE
+    private onResize = (e: DragEvent, col: Col) => {
+      if (this.dragCol != -1 || this.dragRow != -1) return;
+
+      this.isResizing = true;
+
+      const t = e.target as HTMLDivElement;
+      e.dataTransfer?.setDragImage(IMG, 0, 0);
+
+      const elems = this.COLS.map(c => this.shadowRoot!.querySelector<HTMLElement>(`th.col-index-${c.index}`)!);
+      this.COLS.forEach((c, i) => (c.width = elems[i].offsetWidth + "px"));
+      this.updCols();
+
+      const orgX = e.x;
+      this.eX = e.x;
+
+      let posX = -1;
+      let set = -1;
+      let setLeft = -1;
+
+      const colLeft = this.COLS[col.index - 1];
+
+      const elem = elems[col.index];
+      const elemLeft = elems[col.index - 1];
+      const elemW = elem.offsetWidth;
+      const elemLeftW = elemLeft.offsetWidth;
+
+      const drag = () => {
+        const eX = this.eX;
+        if (posX != eX && eX != 0) {
+          posX = eX;
+
+          // TODO - With this I can catch wher resize is stopped
+          /*
+          const real = elem.offsetWidth;
+          console.log(set, real);
+          const realLeft = elemLeft.offsetWidth;
+          console.log(setLeft, realLeft);
+          */
+
+          const deltaX = posX - orgX;
+          set = elemW - deltaX;
+          setLeft = elemLeftW + deltaX;
+
+          col.width = set + "px";
+          colLeft.width = setLeft + "px";
+          this.updCols();
+        }
+      };
+
+      const dragOver = (evt: DragEvent) => (this.eX = evt.x);
+
+      const dragEnd = () => {
+        this.isResizing = false;
+        t.removeEventListener("drag", drag);
+        t.removeEventListener("dragend", dragEnd);
+        document.removeEventListener("dragover", dragOver);
+      };
+
+      t.addEventListener("drag", drag);
+      t.addEventListener("dragend", dragEnd);
+      document.addEventListener("dragover", dragOver);
+    };
+
+    // RENDER
+    // ----------------------------------
+
+    render() {
+      if (this.error)
+        return html`
+          <div>${this.error}</div>
+        `;
+
+      const { head } = this.config;
+      const clazz = classMap({
```

Comment:
lol clazz? nit pic: can we have a descriptive variable name that makes sense and does not remind me of jazz?

Reply:
Done